Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm

```diff
@@ -4013,60 +4013,69 @@ - (pid_t)accessibilityPresenterProcessIdentifier
 - (NSArray *)accessibilityArrayAttributeValues:(NSString *)attribute index:(NSUInteger)index maxCount:(NSUInteger)maxCount
 {
     AXTRACE(makeString("WebAccessibilityObjectWrapper accessibilityArrayAttributeValue:"_s, String(attribute)));
-    RefPtr<AXCoreObject> backingObject = self.updateObjectBackingStore;
-    if (!backingObject)
-        return nil;
 
-    if ([attribute isEqualToString:NSAccessibilityChildrenAttribute]) {
-        const auto& unignoredChildren = backingObject->unignoredChildren();
-        if (unignoredChildren.isEmpty()) {
-            NSArray *children = transformSpecialChildrenCases(*backingObject, unignoredChildren);
-            if (!children)
-                return nil;
+    if ([attribute isEqualToString:NSAccessibilityChildrenAttribute])
+        return [self _accessibilityChildrenFromIndex:index maxCount:maxCount returnPlatformElements:YES];
 
-            NSUInteger childCount = [children count];
-            if (index >= childCount)
-                return nil;
+    return [super accessibilityArrayAttributeValues:attribute index:index maxCount:maxCount];
+}
 
-            NSUInteger arrayLength = std::min(childCount - index, maxCount);
-            return [children subarrayWithRange:NSMakeRange(index, arrayLength)];
-        }
+- (NSArray *)_accessibilityChildrenFromIndex:(NSUInteger)index maxCount:(NSUInteger)maxCount returnPlatformElements:(BOOL)returnPlatformElements
+{
+    if (!returnPlatformElements && !AXObjectCache::clientIsInTestMode())
+        return nil;
```

Comment:
Do we need this guard? Seems like ATs would only hit this if they called this private method, which legitimate ATs would never do. So then the question becomes, could a misbehaving AT be able to do something bad if they call this private method with `returnPlatformElements == NO`?

Reply:
Probably not—this was just to be extra safe. But since this method is private, we can probably remove it.